src/default-config-backend.cpp

```diff
@@ -98,7 +98,7 @@ class dynamic_ini_config_t : public wf::config_backend_t
         config = wf::config::build_configuration(
             get_xml_dirs(), SYSCONFDIR "/wayfire/defaults.ini", config_file);
 
-        int inotify_fd = inotify_init1(IN_CLOEXEC);
+        int inotify_fd = inotify_init1(IN_CLOEXEC | IN_NONBLOCK);
```

Comment:
NAK from me. This should be made to work with blocking reads.

Reply:
How to handle multiple inofity events though?

Reply:
If the event fires, it is readable and we can read at least one event. After that, I would probably use something like `select` to check if there are subsequent events to read from the fd.

Reply:
How would that be any better? Isn't Wayland's event loop callback API supposed to be used like this, as in to return when there's nothing left in the `fd`? That's basically non-blocking, isn't it?

Reply:
> How would that be any better? Isn't Wayland's event loop callback API supposed to be used like this, as in to return when there's nothing left in the `fd`? That's basically non-blocking, isn't it?

Yes you're right, after reading some man pages it's probably better to set the fd nonblocking in this case.

Reply:
but actually, libinotify might be writing while we are reading and this can cause a problem.

Reply:
Are we using `libinotify`?

Reply:
> Are we using `libinotify`?

You're using the linux kernel inotify tools.

Reply:
libinotify is used in FreeBSD IIRC.